### PR TITLE
[Feature] Execute rules and run analytics in msf settings (4)

### LINKED ISSUE #726: New front page (B10)
New front page: a very simplified interface that can be accessed from a dashboard to trigger the synchronization for non-admin users.
- [ ] Logic
   - [x] Smart analytics execution:
      - [x] Run analytics only once if more than one sync rule is executed with analytics enabled
      - [x] Distinguish between local and global instance: If it is an online server (check domain), when ‘run analytics’ settings are disabled, show a message in the UI telling the last execution of analytics.
   - [ ] Button to trigger sync rules will trigger a set of sync rules. The set of sync rules to be synced will be based on an internal metadata rule using the OUs of the user.
   - [x] Merge advance settings with actual sync rules settings: Period for sync rule
   - [x] Merge MSF settings with actual sync rules settings: Run Analytics
- [ ] GUI:
   - [x] ‘Regular’ user
      - [ ] Button ‘Go to History Section’
      - [ ] Button ‘Advance Settings’
image
   - [x] ‘Admin’ user
      - [ ] Button ‘Go to History Section’
      - [ ] Button ‘Go to Admin Dashboard’
      - [ ] Button ‘Advance Settings’
      - [ ] Button ‘MSF Settings’
image
   - [x] Button to trigger sync rules
   - [x] Section displaying progress logs
   - [x] Simplify version of the summary widget
   - [x] Advance Settings dialog
      - [ ] Sync fixed period
image

   - [ ] MSF Settings dialog
      - [ ] Minimum week to get value
      - [x] Run analytics. Three options: true, false and use sync rule setting
         - [x] Online server (check by domain) set to false
         - [x] Offline server set to sync rule setting

### PR DESCRIPTION
**Pull request by xurxodev**
### :pushpin: References

-   **Issue:** Closes #726

### :memo: Implementation

- [x] Add RunAnalyticsSettings to MSFDialog
- [x] Execute event sync rules and show progress
- [x] Create check box in advanced settings to select use sync rules periods
- [x] Update route paths

### :art: Screenshots

Captura de pantalla 2020-12-07 a las 11 20 29
Captura de pantalla 2020-12-07 a las 11 21 12
Captura de pantalla 2020-12-07 a las 11 21 29

### :bookmark_tabs: Others

-   Any change in the [GUI library](https://github.com/EyeSeeTea/d2-ui-components)? If so, what branch/PR?

-   Any change in the [D2 Api](https://github.com/EyeSeeTea/d2-api)? If so, what branch/PR?

### CODE REVIEW

File: src/presentation/webapp/msf-aggregate-data/pages/MSFHomePagePresenter.ts [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+export async function executeAggregateData(
+    api: D2Api,
+    compositionRoot: CompositionRoot,
+    msfSettings: MSFSettings,
+    onProgressChange: (progress: string[]) => void,
+    period?: Period
+) {
+    const eventSyncRules = (
+        await SyncRule.list(api, { type: "events" }, { paging: false })
+    ).objects.slice(0, 2);
```
Comment (SferaDev) [inline src/presentation/webapp/msf-aggregate-data/pages/MSFHomePagePresenter.ts:18]: During the merge I saw this ``slice``, not sure what it does but assuming existence and/or sorting of ``dataStore`` sync rules seems a bit weak to me.

**Comment by SferaDev** [comment]
Bug found with the ``back`` method:

- Go to /#/sync-rules/metadata
- Open edit sync rule
- Go back (accept confirmation dialog)
- Go back (it should go to home page)
- User is back to edit page

File: src/presentation/webapp/msf-aggregate-data/pages/MSFHomePagePresenter.ts (hunk 1 above)
Comment (xurxodev) [reply src/presentation/webapp/msf-aggregate-data/pages/MSFHomePagePresenter.ts:18]: @SferaDev That's is true but It's a temporal implementation agreed with @adrianq

**Comment by xurxodev** [comment]
Solved!

**Pull request merged**